src/transformers/models/rt_detr/modeling_rt_detr.py

```diff
@@ -0,0 +1,1674 @@
+        return output
+
+
+class RepVggBlock(nn.Module):
+    def __init__(self, channels_in, channels_out, activation="relu"):
+        super().__init__()
+        self.channels_in = channels_in
+        self.channels_out = channels_out
+        self.conv1 = RTDetrConvNormLayer(channels_in, channels_out, 3, 1, padding=1, activation=None)
+        self.conv2 = RTDetrConvNormLayer(channels_in, channels_out, 1, 1, padding=0, activation=None)
+        self.activation = nn.Identity() if activation is None else ACT2CLS[activation]()
+
+    def forward(self, x):
+        y = self.conv1(x) + self.conv2(x)
+        return self.activation(y)
+
+
+class RTDetrCSPRepLayer(nn.Module):
+    def __init__(self, config: RTDetrConfig):
+        super().__init__()
+
+        in_channels = config.hidden_dim * 2
+        out_channels = config.hidden_dim
+        num_blocks = round(3 * config.depth_mult)
+        activation = config.act_encoder
+
+        hidden_channels = int(out_channels * config.expansion)
+        self.conv1 = RTDetrConvNormLayer(in_channels, hidden_channels, 1, 1, bias=None, activation=activation)
+        self.conv2 = RTDetrConvNormLayer(in_channels, hidden_channels, 1, 1, bias=None, activation=activation)
+        self.bottlenecks = nn.Sequential(
+            *[RepVggBlock(hidden_channels, hidden_channels, activation=activation) for _ in range(num_blocks)]
+        )
+        if hidden_channels != out_channels:
+            self.conv3 = RTDetrConvNormLayer(hidden_channels, out_channels, 1, 1, bias=None, activation=activation)
+        else:
+            self.conv3 = nn.Identity()
+
+    def forward(self, x):
+        x_1 = self.conv1(x)
+        x_1 = self.bottlenecks(x_1)
+        x_2 = self.conv2(x)
+        return self.conv3(x_1 + x_2)
+
+
+class RTDetrMSDeformableAttention(nn.Module):
+    def __init__(self, config: RTDetrConfig):
+        """
+        Multi-Scale Deformable Attention Module
+        """
+        super().__init__()
+
+        self.embed_dim = config.hidden_dim
+        self.num_heads = config.num_attention_heads
+        self.num_levels = config.num_levels
+        self.num_points = config.num_decoder_points
+        self.total_points = self.num_heads * self.num_levels * self.num_points
+        self.head_dim = self.embed_dim // self.num_heads
+
+        if self.head_dim * self.num_heads != self.embed_dim:
+            raise ValueError("Relation self.head_dim * num_heads == self.embed_dim does not apply")
+        self.sampling_offsets = nn.Linear(self.embed_dim, self.total_points * 2)
+        self.attention_weights = nn.Linear(self.embed_dim, self.total_points)
+        self.value_proj = nn.Linear(self.embed_dim, self.embed_dim)
+        self.output_proj = nn.Linear(self.embed_dim, self.embed_dim)
+        self.ms_deformable_attn_core = deformable_attention_core_func
+
+    def forward(self, query, reference_points, value, value_spatial_shapes, value_mask=None):
+        bs, len_q = query.shape[:2]
```

Comment:
```suggestion
        batch_size, seq_length = query.shape[:2]
```
let's use clearer variable names